docs/error-codes.md

```diff
@@ -1569,6 +1569,10 @@ void TestMethod(MethodInfo methodInfo)
 }
 ```
 
+#### `IL2104`: Assembly 'assembly' produced trim warnings
```

Comment:
@samsp-msft This change is making it so that app authors will get at most one trim analysis warning from nuget libraries by default. PTAL at the wording of the warning message and aka.ms link, currently:
```
`IL2104`: Assembly 'assembly' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries
```
The link will point to docs that:
- recommend the app author contact the library author about making the library trim-friendly
- instructions for library authors to make their libraries trim-friendly

Note we have an existing link, https://aka.ms/dotnet-illink, which is the landing page for the linker docs, hence the "/libraries".